### PR TITLE
Support constant evaluation for `#if` and `#elif` preprocessor directive

### PR DESCRIPTION
**Pull request by ChAoSUnItY**
- Implement constant evaluation for `#if` and `#ifdef` preprocessor directive
  - Notice that short-circuit is not yet implemented.

### CODE REVIEW

**Comment by ChAoSUnItY** [comment]
This Pull Request will be ready for review once after #111 is merged.

**Review by jserv — CHANGES_REQUESTED** [review]
Rebase the latest `master` branch.

**Comment by jserv** [comment]
I defer to @vacantron for confirmation.

**Review by vacantron — APPROVED** [review]

**Comment by jserv** [comment]
Thank @ChAoSUnItY for contributing!

**Pull request merged**
